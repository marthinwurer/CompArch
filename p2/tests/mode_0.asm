;
; tests addressing mode 0
;
; uses instructions HALT, CLR, BPT
;
; uses no other addressing modes
;
	.globl	start
done:	halt
;
. = . + 100
;
start:	mov	r7,r2
	bpt
	clr	r7
	bpt
;
	.end	start
