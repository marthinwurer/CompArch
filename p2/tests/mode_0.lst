      1                                 ;
      2                                 ; tests addressing mode 0
      3                                 ;
      4                                 ; uses instructions HALT, CLR, BPT
      5                                 ;
      6                                 ; uses no other addressing modes
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000000                  done:    halt
     10                                 ;
     11 000102                          . = . + 100
     12                                 ;
     13 000102  010702                  start:   mov     r7,r2
     14 000104  000003                           bpt
     15 000106  005007                           clr     r7
     16 000110  000003                           bpt
     17                                 ;
     18 000112  000102'                          .end    start


done     000000 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000102 RG 
