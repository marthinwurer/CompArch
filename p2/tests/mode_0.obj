000 2 00 00
042 2 c2 11
044 2 03 00
046 2 07 0a
048 2 03 00
042
