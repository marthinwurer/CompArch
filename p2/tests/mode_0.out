CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 102
000102:  PS[00]  010702  MOV R7,R2
000104:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000104]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000106]
000106:  PS[00]  005007  CLR R7
000000:  PS[04]  000000  HALT

Machine Halted - HALT instruction

Simulated time 19 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
